### PR TITLE
Console support for Submariner air-gapped clusters

### PR DESCRIPTION
**Pull request by sridhargaddam**
This PR includes the necessary changes in the ACM console to allow users to install Submariner on air-gapped clusters.

Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

**Comment by sridhargaddam** [comment]
Sample output:

Screenshot from 2022-11-25 15-14-30

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetDetails.test.tsx [hunk 1]
```diff
@@ -833,6 +840,7 @@ const mockManagedClusterNoCredentialsSubmarinerConfigOpenstack: SubmarinerConfig
             },
         },
         IPSecNATTPort: submarinerConfigDefault.nattPort,
+        airGappedDeployment: submarinerConfigDefault.airGappedDeployment,
```
Comment (aswinsuryan) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetDetails.test.tsx:843]: In one test you can use the non-default value.

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx [hunk 2]
```diff
@@ -915,6 +919,23 @@ export function InstallSubmarinerForm(props: { availableClusters: Cluster[] }) {
                                         setOpenStackInstanceTypes(copy)
                                     },
                                 },
+                                {
+                                    id: 'isAirGappedDeployment',
```
Comment (yboaron) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:923]: Is it possible to run air-gapped deployment for managed OCP (ARO,ROSA) ? 

if it isn't you should make this field as hidden for managed services

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx [hunk 3]
```diff
@@ -915,6 +919,23 @@ export function InstallSubmarinerForm(props: { availableClusters: Cluster[] }) {
...
+                                    type: 'Checkbox',
+                                    label: t('Air-gapped Cluster'),
+                                    labelHelp: t(
+                                        'Enable this option to instruct Submariner not to access any external servers for public-ip resolution.'
```
Comment (skitt) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:927]:
```suggestion
                                        'Enable this option to instruct Submariner not to access any external servers for public IP resolution.'
```

**Review by skitt — CHANGES_REQUESTED** [review]
I was hoping there would be a centralised “air gap” setting common to the ACM or OpenShift installation, but I haven’t found one.

Let’s change “public-ip” to “public IP”.

File: frontend/public/locales/en/translation.json [hunk 4]
```diff
@@ -1330,6 +1331,7 @@
     "Enable policy": "Enable policy",
     "Enable resource editing": "Enable resource editing",
     "Enable search to display application statuses properly.": "Enable search to display application statuses properly.",
+    "Enable this option to instruct Submariner not to access any external servers for public-ip resolution.": "Enable this option to instruct Submariner not to access any external servers for public-ip resolution.",
```
[the remark this answers is not in the document]
Comment (sridhargaddam) [reply frontend/public/locales/en/translation.json:1334]: We used `public-ip` in our [release-notes](https://submariner.io/community/releases/), so followed the same convention. Anyways, I can change it.

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx (hunk 2 above)
Comment (sridhargaddam) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:923]: I have to check the general ACM behavior when it comes to Air-Gapped clusters - i.e. if they have a flag/config-param to convey that deployment is on air-Gapped clusters and if they restrict usage of that flag based on the cluster location. In general AFAICT, most of the air-gapped clusters are onPrem clusters and even if they use public clouds, the cluster VPC might be configured to block external access. Let me find more details about this. I want to handle this in a follow-up PR, hope its okay for you?

**Comment by sridhargaddam** [comment]
@aswinsuryan haha.. 100% code coverage now :D

**Review by skitt — DISMISSED** [review]

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx (hunk 2 above)
Comment (yboaron) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:923]: Well, it seems that air-gapped/disconnected deployments are supported also for managed OCP, check [1].

Didn't find a central airgap/disconnected flag either, I can see that for ACM create-cluster (OSP or VMWARE) there is a disconnected section [2] that is copied into OCP's install-config **imageContentSources** field.


But yeah, sure, we can handle it in a follow-up PR if needed.

[1] 
https://github.com/redhat-cop/ocp-disconnected-docs/blob/main/AROInstall.md
[2]
image

**Review by yboaron — DISMISSED** [review]

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx (hunk 2 above)
Comment (sridhargaddam) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:923]: thanks for checking @yboaron

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetDetails.test.tsx [hunk 5]
```diff
@@ -1018,6 +1026,7 @@ describe('ClusterSetDetails page', () => {
         await waitForTestId('credential-secret')
         await waitForNotTestId('awsAccessKeyID')
         await waitForNotTestId('awsSecretAccessKeyID')
+        await clickByLabel('Air-gapped Cluster')
```
Comment (nyechiel) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetDetails.test.tsx:1029]: Probably a question for the console team, but is this the right official term we should be using? At least the docs refer to "disconnected installation", but I am not sure what the guidance for this is.

**Review by aswinsuryan — DISMISSED** [review]

**Comment by nyechiel** [comment]
@KevinFCormier can you review please?

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetDetails.test.tsx (hunk 5 above)
Comment (nyechiel) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetDetails.test.tsx:1029]: Based on this [thread](https://coreos.slack.com/archives/CUHCCV19P/p1669822206262309) it looks like disconnected is a better term. cc @oafischer

Comment (KevinFCormier) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetDetails.test.tsx:1029]: Also, only the first word in a heading is capitalized.

Comment (sridhargaddam) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetDetails.test.tsx]: Thanks @nyechiel @KevinFCormier, updated the label to "Disconnected cluster"

**Pull request merged**